sdk/azcore/runtime/policy_bearer_token.go

```diff
@@ -93,5 +94,31 @@ func (b *BearerTokenPolicy) Do(req *policy.Request) (*http.Response, error) {
 			}
 		}
 	}
-	return res, err
+	return res, ensureNonRetriable(err)
+}
+
+func ensureNonRetriable(err error) error {
+	if err != nil {
+		if _, ok := err.(errorinfo.NonRetriable); !ok {
+			err = btpError{err}
+		}
+	}
+	return err
+}
+
+// btpError is a wrapper that ensures RetryPolicy doesn't retry requests BearerTokenPolicy couldn't authorize
+type btpError struct {
+	err error
```

Comment:
If you anonymously embed `error` then you can remove the `Error()` and `Unwrap()` methods.

We should put this into `sdk/internal` at some point as we have several copies of this floating around.

Reply:
Thanks for the embedding tip. I opened #19467 to track moving this into `internal`.